crates/router/src/core/payment_methods/transformers.rs

```diff
@@ -1,104 +1,220 @@
-    utils::{self, OptionExt},
+    utils,
 };
 
 #[derive(Debug, Deserialize, Serialize)]
-#[serde(rename_all = "camelCase")]
-pub struct AddCardRequest<'a> {
+pub struct StoreCardReq<'a> {
+    pub merchant_id: &'a str,
+    pub merchant_customer_id: String,
+    pub card: Card,
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+pub struct Card {
     pub card_number: Secret<String, pii::CardNumber>,
-    pub customer_id: &'a str,
+    pub name_on_card: Option<Secret<String>>,
     pub card_exp_month: Secret<String>,
     pub card_exp_year: Secret<String>,
-    pub merchant_id: &'a str,
-    pub email_address: Option<Secret<String, pii::Email>>,
-    pub name_on_card: Option<Secret<String>>,
-    pub nickname: Option<String>,
+    pub card_brand: Option<String>,
+    pub card_isin: Option<String>,
+    pub nick_name: Option<String>,
 }
 
 #[derive(Debug, Deserialize, Serialize)]
-#[serde(rename_all = "camelCase")]
-pub struct GetCard<'a> {
-    merchant_id: &'a str,
-    card_id: &'a str,
+pub struct StoreCardResp {
+    pub status: String,
+    pub error_message: Option<String>,
+    pub error_code: Option<String>,
+    pub payload: Option<StoreCardRespPayload>,
 }
 
 #[derive(Debug, Deserialize, Serialize)]
-#[serde(rename_all = "camelCase")]
-pub struct AddCardResponse {
-    pub card_id: String,
-    pub external_id: String,
-    pub card_fingerprint: Secret<String>,
-    pub card_global_fingerprint: Secret<String>,
-    #[serde(rename = "merchant_id")]
-    pub merchant_id: Option<String>,
-    pub card_number: Option<Secret<String, pii::CardNumber>>,
-    pub card_exp_year: Option<Secret<String>>,
-    pub card_exp_month: Option<Secret<String>>,
-    pub name_on_card: Option<Secret<String>>,
-    pub nickname: Option<String>,
-    pub customer_id: Option<String>,
-    pub duplicate: Option<bool>,
+pub struct StoreCardRespPayload {
+    pub card_reference: String,
 }
 
 #[derive(Debug, Deserialize, Serialize)]
-pub struct GetCardResponse {
-    pub card: AddCardResponse,
+pub struct CardReqBody<'a> {
+    pub merchant_id: &'a str,
+    pub merchant_customer_id: String,
+    pub card_reference: String,
 }
 
 #[derive(Debug, Deserialize, Serialize)]
-#[serde(rename_all = "camelCase")]
-pub struct DeleteCardResponse {
-    pub card_id: Option<String>,
-    pub external_id: Option<String>,
-    pub card_isin: Option<Secret<String>>,
+pub struct RetrieveCardResp {
     pub status: String,
+    pub error_message: Option<String>,
+    pub error_code: Option<String>,
+    pub payload: Option<RetrieveCardRespPayload>,
 }
 
-pub fn mk_add_card_request(
+#[derive(Debug, Deserialize, Serialize)]
+pub struct RetrieveCardRespPayload {
+    pub card: Option<Card>,
+    pub enc_card_data: Option<String>,
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+pub struct DeleteCardResp {
+    pub status: String,
+    pub error_message: Option<String>,
+    pub error_code: Option<String>,
+}
+
+pub fn basilisk_hs_key_id() -> &'static str {
+    "1"
+}
```

Comment:
Why is this hard coded? can we fetch it fron configuration?

Reply:
hardcoded in basilisk-hs as well , will keep in sync with basilisk